aiida_mlip/data/model.py

```diff
@@ -0,0 +1,125 @@
+                    file.unlink()
+                    return existing_file, architecture
+
+                if arch_path == cache_path and arch_path != ex_file_path:
+                    architecture = ex_file_path.name
+                    file.unlink()
+                    return existing_file, architecture
+    return file, architecture
+
+
+class ModelData(SinglefileData):
+    """Class to handle models"""
+
+    def __init__(self, file, filename, architecture: str = None, **kwargs):
+        """Add a file to the node, parse it and set the attributes found.
+
+        :param file: absolute path to the file or a filelike object
+        :param filename: specify filename to use (defaults to name of provided file).
+        :param architecture: specify architecture
+        """
+
+        super().__init__(file, filename, **kwargs)
+        self.base.attributes.set("architecture", architecture)
+
+    def set_file(self, file, filename=None, architecture: str = None, **kwargs):
+        """Add a file to the node, parse it and set the attributes found.
+
+        :param file: absolute path to the file or a filelike object
+        :param filename: specify filename to use (defaults to name of provided file).
+        """
+
+        super().set_file(file, filename, **kwargs)
+
+        self.base.attributes.set("architecture", architecture)
+
+    @classmethod
+    def local_file(cls, file: str, filename: str = None, architecture: str = None):
+        """Sve a local file as ModelData"""
+        file_path = Path(file).resolve()
+        return cls(file=str(file_path), filename=filename, architecture=architecture)
+
+    @classmethod
+    def download(
+        cls,
+        url: str,
+        filename: str = None,
+        cache_dir: str = None,
+        architecture: str = None,
+    ):
+        """Download a file from a URL and save it as a ModelData"""
+        if not cache_dir:
+            cache_dir = "~/.cache/mlips/"
+        if architecture is None:
+            arch_dir = Path(cache_dir)
+        else:
+            arch_dir = Path(cache_dir).joinpath(architecture)
+
+        cache_path = Path(cache_dir).resolve()
+        arch_path = arch_dir.resolve()
+        arch_path.mkdir(parents=True, exist_ok=True)
+
+        model_name = urlparse(url).path.split("/")[-1]
+
+        if filename:
+            file = arch_path.joinpath(filename)
+        else:
+            file = arch_path.joinpath(model_name)
+
+        # Check if there is already a file named that way and rename it
+        if file.exists():
+            file = file.with_name(file.stem + "_2" + file.suffix)
+            # Check again if the new filename already exists, just in case
+            while file.exists():
+                # If the new filename also exists, keep incrementing the number
+                file = file.with_name(
+                    file.stem[:-2] + str(int(file.stem[-2:]) + 1).zfill(2) + file.suffix
+                )
```

Comment:
`with_stem` is Python 3.9+, otherwise can use `with_name` in similar way.
```suggestion
        stem = file.stem
        i = 1
        while file.exists():
            i += 1
            file = file.with_stem(f"{stem}_{i}")
```